### PR TITLE
Add test framework

### PR DESCRIPTION
**Pull request by ath88**
Check out #9 first

### CODE REVIEW

File: lib/circles/create.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ module.exports = function(req, res) {
         mission: req.body.mission || null,
         aim: req.body.aim || null,
         expectationsForMembers: [],
-        members: [req.user.id],
+        members: [req.user.userId],
         invited: req.body.invited || [],
-        contactPerson: req.user.id,
+        contactPerson: req.user.userId,
```
Comment (hypesystem) [inline lib/circles/create.js]: I'm not sure I like this pattern. req.user.id adds plenty of context to the id, but when documenting it, it makes sense to be explicit.

File: lib/users/create.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ function createUser(pgdb, remoteId, name, callback) {
             const user = {
                 token: result.rows[0].token,
                 name: result.rows[0].data.name,
-                id: result.rows[0].data.remoteId
+                userId: result.rows[0].data.userId
```
Comment (hypesystem) [inline lib/users/create.js:30]: it makes sense that we lose the "remote" part, though :smile:

File: features/createUser.feature [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Feature: Create user
+  As an external brand user
+  I want to be able to retrieve a token for a user
+  So that the user will be able to use the API
+
+  Scenario: Creating user
+    Given I have a users name
+        And I have a users id
```
Comment (hypesystem) [inline features/createUser.feature:8]: Odd indentation?

**Review by hypesystem — APPROVED** [review]
Nice with the test harness :smile:

File: lib/circles/create.js (hunk 1 above)
Comment (ath88) [reply lib/circles/create.js]: I generally prefer clarity over brevity. 

https://signalvnoise.com/posts/3250-clarity-over-brevity-in-variable-and-method-names

Consider the case where per chance the user-object was put in a variable called `current`. In that case, we would be referring to `current.id` or `current.userId`. I consider it the clarity a decent tradeoff for the brevity. :)

You could always argue that the user object should never be called `current`, but be called `currentUser`, but this way we prevent that from ever becoming an issue.

File: features/createUser.feature (hunk 3 above)
Comment (ath88) [reply features/createUser.feature:8]: On purpose. The `And` is also a 'Given`. I am not too used to writing gherkin though.

**Pull request merged**